app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,50 @@
+class OrderItemsController < ApplicationController
+  before_action :current_user
+
+  def create
+    my_order
+    @order.save
+    cookies.signed[:order] = @order.id
+
+    item = @order.order_items.where(product_id: params[:product_id]).first
+
+    unless item.nil?
+      item.update(quantity: order_item_params[:quantity])
+    else
+      @order.order_items << item = OrderItem.create(order_item_params)
+    end
+
+    redirect_to cart_path
+  end
+
+  def cart
+    @instock = my_order.instock
+    @outofstock = my_order.outofstock
+    @total = my_order.cart_total
+  end
```

Comment:
Since these methods are defined in the Order Model and this is the `OrderItemsController`, it is best to put a comment about where the functionality can be found